docs: release process: corrections and clarifications (#108572)
zephyrproject-rtos/zephyr: doc/project/dev_env_and_tools.rst

     - Description

   * - :guilabel:`priority: {high|medium|low}`

>      - To classify the impact and importance of an issue.

Reviewer: ```suggestion
     - To classify the impact and importance of a bug.
``` 
we don't add priority to "issues" in general only bugs.
(I don't know why `feature`s was listed here, but we do not add priority to them)
